packages/@react-aria/datepicker/src/useDateSegment.ts

```diff
@@ -86,6 +85,21 @@ export function useDateSegment<T extends DateValue>(props: DatePickerProps<T> &
 
   let parser = useMemo(() => new NumberParser(locale, {maximumFractionDigits: 0}), [locale]);
 
+  let backspace = () => {
+    if (parser.isValidPartialNumber(segment.text) && !props.isReadOnly && !segment.isPlaceholder) {
+      let newValue = segment.text.slice(0, -1);
+      let parsed = parser.parse(newValue);
+      if (newValue.length === 0 || parsed === 0) {
+        state.clearSegment(segment.type);
+      } else {
```

Comment:
What do you think about also focusing the previous segment after clearing the current one? Mirrors how we advance to the next segment once a valid value is typed in the current one

Reply:
On second thought this might be fine, a user would probably not want focus to leave the segment they just cleared so that they can enter a new value

Reply:
Yeah that was the reason I didn't do it